packages/async-flow/src/convert.js

```diff
@@ -0,0 +1,131 @@
+import { Fail, X, annotateError, makeError, q } from '@endo/errors';
+import { throwLabeled } from '@endo/common/throw-labeled.js';
+import {
+  getErrorConstructor,
+  getTag,
+  isObject,
+  makeTagged,
+  passStyleOf,
+} from '@endo/pass-style';
+import { isVow } from '@agoric/vow/src/vow-utils.js';
+import { objectMap } from '@endo/common/object-map.js';
+
+const makeConvert = (convertRemotable, convertPromiseOrVow, convertError) => {
+  const convertRecur = (specimen, label) => {
+    // Open code the synchronous part of applyLabelingError, because
+    // we need to preserve returned promise identity.
+    // TODO switch to Richard Gibson's suggestion for a better way
+    // to keep track of the error labeling.
+    // See https://github.com/endojs/endo/pull/1795#issuecomment-1756093032
+    if (label === undefined) {
+      // eslint-disable-next-line no-use-before-define
+      return innerConvert(specimen);
+    }
+    try {
+      // eslint-disable-next-line no-use-before-define
+      return innerConvert(specimen);
+    } catch (err) {
+      throwLabeled(err, label);
+    }
+  };
+
+  const innerConvert = specimen => {
+    if (!isObject(specimen)) {
+      return specimen;
+    }
+    const passStyle = passStyleOf(specimen);
+    switch (passStyle) {
+      case 'copyArray': {
+        return specimen.map((element, i) => convertRecur(element, i));
+      }
+      case 'copyRecord': {
+        return objectMap(specimen, (value, name) => convertRecur(value, name));
+      }
+      case 'tagged': {
+        if (isVow(specimen)) {
+          return convertPromiseOrVow(specimen);
+        }
+        const tag = getTag(specimen);
```

Comment:
Genuine bug I was trying to get the tag with `specimen.tag`, rather than the `getTag` function. Found by testing. Fixed.